components/AvalancheForecastZoneMap.tsx

```diff
@@ -92,13 +102,112 @@ export const CARD_WHITESPACE = (1 - CARD_WIDTH) / 2; // proportion of overall wi
 export const CARD_SPACING = CARD_WHITESPACE / 2; // proportion of overall width that the spacing between two cards takes up
 export const CARD_MARGIN = CARD_SPACING / 2; // proportion of overall width that each card needs as a margin
 
+enum AnimatedDrawerState {
+  Hidden = 'Hidden',
+  Docked = 'Docked',
+  Visible = 'Visible',
+}
+
+class AnimatedDrawerController {
+  // These offsets are applied through translateY on the FlatList
+  static readonly OFFSETS = {
+    [AnimatedDrawerState.Hidden]: 160,
+    [AnimatedDrawerState.Docked]: 120,
+    [AnimatedDrawerState.Visible]: 0,
+  };
+
+  // When a pan gesture goes beyond this distance, we animate to the final state
+  static readonly SNAP_THRESHOLD = 16;
+
+  state: AnimatedDrawerState;
+  baseOffset: number;
+  panning: boolean;
+  yOffset: Animated.Value;
+
+  constructor(state = AnimatedDrawerState.Docked) {
+    this.state = state;
+    this.baseOffset = AnimatedDrawerController.OFFSETS[state];
+    this.panning = false;
+    this.yOffset = new Animated.Value(this.baseOffset);
+  }
+
+  setState(state: AnimatedDrawerState) {
+    this.state = state;
+    this.panning = false;
+    this.baseOffset = AnimatedDrawerController.OFFSETS[state];
+    this.yOffset.setOffset(this.baseOffset);
+    this.yOffset.setValue(0);
+  }
+
+  onPanResponderGrant() {
+    this.panning = true;
+    this.yOffset.setOffset(this.baseOffset);
+    this.yOffset.setValue(0);
+  }
+
+  onPanResponderMove(event: GestureResponderEvent, gestureState: PanResponderGestureState) {
+    if (!this.panning) {
+      return;
+    }
+
+    // Are we moving too far in the X direction? If so, treat as a scroll and stop panning the drawer
+    if (Math.abs(gestureState.dx) > AnimatedDrawerController.SNAP_THRESHOLD) {
+      this.onPanResponderRelease();
+      return;
+    }
+
+    // Detect overscroll in the invalid direction - we allow a little bit of give,
```

Comment:
Might be overkill but I feel like I've seen apps where the animation "decelerates" when being pushed out too far from the bounds, which makes this feel less bouncy.

Reply:
I'd like to get this in the designers' hands and see if it's what they were expecting - they may want something more like easing in/out, but it's pretty easy to tweak later. I didn't have a motion example to work off of.